metadata_layer.py

```diff
@@ -791,12 +795,12 @@ def mirror_link(self, model_object, data_source, link, link_obj, policy):
         link_obj.resource.representation = representation
 
         # If we couldn't fetch this representation, don't mirror it,
-        # and if this was an open access link, then suppress the associated
+        # and if this was an open/protected access link, then suppress the associated
         # license pool until someone fixes it manually.
         # The license pool to suppress will be either the passed-in model_object (if it's of type pool),
         # or the license pool associated with the passed-in model object (if it's of type edition).
         if representation.fetch_exception:
-            if pools and link.rel == Hyperlink.OPEN_ACCESS_DOWNLOAD:
+            if pools and link.rel in [Hyperlink.OPEN_ACCESS_DOWNLOAD, Hyperlink.GENERIC_OPDS_ACQUISITION]:
```

Comment:
This code is better than the alternative but I don't think it will ever be activated. It looks like this code is using `Representation.get` to retrieve a book by its URL before mirroring. I wouldn't expect that to work for books that are non-open-access; instead I'd expect the book to be retrieved some other way and stored in LinkData.content. Am I missing something?

Reply:
I reverted my changes in both places in this method. Are there any other changes required?